src/errorReportingService.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  DriverConfig,
+  ErrorReportingConfig,
+  ErrorReportingServiceContract,
+} from '@ioc:Adonis/Addons/Zeytech/ErrorReporting'
+import { LoggerContract } from '@ioc:Adonis/Core/Logger'
+import { ConfigContract } from '@ioc:Adonis/Core/Config'
+import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import { Driver } from './driver'
+
+export class ErrorReportingService implements ErrorReportingServiceContract {
+  private drivers: Driver[] = []
+
+  constructor(private logger: LoggerContract, config: ConfigContract) {
+    const errorReportingConfig = config.get('zeytech-error-reporting') as ErrorReportingConfig
+
+    for (const driverConfig of errorReportingConfig.drivers) {
```

Comment:
Let's move this logic to an async init function.  Async work in a constructor isn't recommended.  You could do something like:

```
const promises = errorReportingConfig.drivers.map(async (driverConfig) => {
  return await this.loadDriver(driverConfig)
})
return await Promise.all(promises)
```

Then you can make the boot function in the provider async and call `await ErrorReportingService.init()` in there.